sklearn/pipeline.py

```diff
@@ -20,12 +21,73 @@
 from .externals import six
 from .utils import tosequence
 from .utils.metaestimators import if_delegate_has_method
-from .externals.six import iteritems
 
 __all__ = ['Pipeline', 'FeatureUnion']
 
 
-class Pipeline(BaseEstimator):
+class _BasePipeline(six.with_metaclass(ABCMeta, BaseEstimator)):
```

Comment:
Very possibly. I'm not well acquainted and will have a look. Do we think it'd be especially useful there?